### ISSUE TITLE
Combine IPython and jedi completion logic

### ISSUE THREAD

**Issue by martinRenou** [ipython/ipython#11673]
We are trying to implement traits autocompletion in `traitlets`, (see [this PR](https://github.com/ipython/traitlets/pull/515)). We achieve this by overwriting the `__init__` method `__signature__`. It works just fine when `jedi` is not enabled. The thing is that does not work with `jedi`  (see [this issue](https://github.com/davidhalter/jedi/issues/1058)).

Until it is fixed in `jedi`, I guess we could fix it in IPython by taking into account both the completion from `jedi` and the completion computed in IPython (because IPython autocompletion logic looks like it takes the `__signature__` into account?).

**Comment by Carreau** [comment]
There is a longstanding plan to try to get all the IPython features into Jedi –  I'd love to have funding to push Jedi forward. So yes this would be great. I'm also trying to get funding to get @LucianaMarques for this winter outreachy, I believe she is interesting in doing some work on the completer.

**Comment by LucianaMarques** [comment]
Hey @martinRenou ! I opened issue #11622 a while ago concerning some improvements I wanted to do to the completer.py file as a whole (not only one class). I would like to help you if you want and maybe we could work on some improvements together, perhaps open a pull request in which we both push some commits. Tell me what you think and we can arrange a screen call or something like that! :)

**Comment by LucianaMarques** [comment]
I'm still getting to know the whole project system design, but I believe that changing too many things at once can break code. We need to think about possible tests.

**Comment by martinRenou** [comment]
Sure! I would be happy to collaborate and contribute a bit to the IPython completion logic, I feel like it deserves a bit more love at some places :)

I wanted to try and fix [this other issue](https://github.com/ipython/ipython/issues/11530), if you want to take a look at it. It seems like people are interested in having **only** path completion when typing in a string, and that [this part of the code](https://github.com/ipython/ipython/blob/65778adec13a65f82c5d2bab123e7a63516702c8/IPython/core/completer.py#L1371-L1377) is supposed to achieve this but does not work.

> I'd love to have funding to push Jedi forward

It would be awesome.

**Comment by LucianaMarques** [comment]
> Sure! I would be happy to collaborate and contribute a bit to the IPython completion logic, I feel like it deserves a bit more love at some places :)

Yup, me too.

> I wanted to try and fix [this other issue](https://github.com/ipython/ipython/issues/11530), if you want to take a look at it. It seems like people are interested in having **only** path completion when typing in a string, and that [this part of the code](https://github.com/ipython/ipython/blob/65778adec13a65f82c5d2bab123e7a63516702c8/IPython/core/completer.py#L1371-L1377) is supposed to achieve this but does not work.

Seems interesting, this should be something we could work on. I'll write you an e-mail and we can schedule a call!

**Comment by flying-sheep** [comment]
Hi! Is there any progress here? I hate to be indirectly responsible (I did the PR to reactivate jedi) for people having problems with the completion, so I’m very interested in getting this fixed :smile:

**Comment by martinRenou** [comment]
Hi :) No progress yet, I don't really have time to take a look at it... I have some ideas for fixing this issue https://github.com/ipython/ipython/issues/11530#issuecomment-477150494. One idea would be to disable jedi autocompletion when typing in a string. I guess that could be a good start for making people happy.

> I hate to be indirectly responsible (I did the PR to reactivate jedi) for people having problems with the completion

I won't blame you for reactivating jedi, jedi is really good ;)

**Comment by flying-sheep** [comment]
> One idea would be to disable jedi autocompletion when typing in a string. I guess that could be a good start for making people happy.

definitely, and as you said, there’s some code that needs to be fixed to have that and make people happy.

**Comment by LucianaMarques** [comment]
Hey guys! 

I have been extremely busy for the past few weeks. I did, however, submit a PR, in which some tests are still not passing but I'll fix it as soon as possible.

I see that a lot of effort is being put into the completion feature. I think we could try to communicate more about it, because if every one of us work independently in a part of this, work will probably not be in sync or time could be wasted in unnecessary code. 

What do you guys think?

@martinRenou I apologize for not answering your e-mail, as I said I am very busy right now but I'll come back to you in a few days.
